temp/OnnxTransformTempSolution/Microsoft.ML.Transforms.Onnx/OnnxTransform.cs

```diff
@@ -0,0 +1,294 @@
+            public string ModelFile;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 1)]
+            public string InputColumn;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 2)]
+            public string OutputColumn;
+
+            public OnnxModelInfo ModelInfo;
+        }
+
+        public const string Summary = "Transforms the data using the Onnx model.";
+        public const string UserName = "OnnxTransform";
+        public const string ShortName = "Onnx";
+        private const string RegistrationName = "OnnxTransform";
+
+        /// <summary>
+        /// Convenience constructor for public facing API.
+        /// </summary>
+        /// <param name="env">Host Environment.</param>
+        /// <param name="input">Input <see cref="IDataView"/>. This is the output from previous transform or loader.</param>
+        /// <param name="modelFile">This is the frozen Onnx model file. https://www.tensorflow.org/mobile/prepare_models </param>
+        /// <param name="name">Name of the output column. Keep it same as in the Onnx model.</param>
+        /// <param name="source">Name of the input column(s). Keep it same as in the Onnx model.</param>
+        public static IDataTransform Create(IHostEnvironment env, IDataView input, string modelFile, string inputColumn, string outputColumn)
+        {
+            return Create(env, new Arguments() { ModelFile = modelFile, InputColumn = inputColumn, OutputColumn = outputColumn }, input);
+        }
+
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            Contracts.CheckValue(env, nameof(env));
+            var host = env.Register(RegistrationName);
+            host.CheckValue(args, nameof(args));
+            host.CheckNonWhiteSpace(args.InputColumn, nameof(args.InputColumn));
+            host.CheckNonWhiteSpace(args.OutputColumn, nameof(args.OutputColumn));
+            host.CheckNonWhiteSpace(args.ModelFile, nameof(args.ModelFile));
+            host.CheckUserArg(File.Exists(args.ModelFile), nameof(args.ModelFile));
+
+            var mapper = new OnnxMapper(host, input.Schema, args);
+            return new RowToRowMapperTransform(host, input, mapper);
+        }
+
+        internal sealed class OnnxMapper : IRowMapper
+        {
+            private readonly IHost _host;
+            private readonly Arguments _args;
+            private OnnxModel _model;
+
+            private readonly OnnxNodeInfo _outputNodeInfo;
+            private readonly ColumnType _outputColType;
+            private readonly string _outputColName;
+
+            private readonly IdvToTensorAdapter _idvToTensorAdapter;
+
+            public const string LoaderSignature = "OnnxMapper";
+            private static VersionInfo GetVersionInfo()
+            {
+                return new VersionInfo(
+                    modelSignature: "ONNX",
```

Comment:
Signatures need to be 8 characters long.

Reply:
What's your suggestion?
ONNXTRFM
ONNXTSFM
ONNXTRAN

Reply:
Perhaps "ONNXSCOR" ?

---
In reply to: [215616699](https://github.com/shmoradims/machinelearning/pull/1#discussion_r215616699) [](ancestors = 215616699)